tests/suites/test_suite_psa_crypto_driver_wrappers.function

```diff
@@ -923,7 +923,19 @@ void export_key(int force_status_arg,
 
     if (PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(output_key_type) &&
         !PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(input_key_type)) {
+#if defined(MBEDTLS_AES_C)
         TEST_EQUAL(mbedtls_test_driver_key_management_hooks.hits, 1);
+#else
+        /* Successfull and non-fake key export require RNG. If CTR_DRBG uses
+         * PSA for AES-ECB, then these random calls cause extra driver access
+         * which we must take into account here. In case of failure, there
+         * is no extra RNG requirement/access. */
+        if ((fake_output->len == 0) && (expected_status == PSA_SUCCESS)) {
+            TEST_EQUAL(mbedtls_test_driver_key_management_hooks.hits, 3);
```

Comment:
Counting the number of expected hits from running CTR_DRBG for the sake of the PSA RNG instance is fragile: it depends on the details of how CTR_DRBG works and on what the cryptographic code does. The RNG is used for blinding and this can vary a lot. It is not reasonable to expect that if blinding changes slightly then completely unrelated test code needs to change. I think the count here is in fact wrong because it depends on the curve of the key: it only currently happens that we only test the driver wrappers with secp256r1. And the count actually depends on the RNG output (because the RNG is used to generate a number in the range [1,p-1]) — it's only passing with a very high probability, not guaranteed (p is almost 2^256 so the chance that the 256 bits = 2 blocks of CTR_DRBG output represent a number outside the desired range is very small).

The conditions for this are also fragile: you need to know that the PSA RNG is using CTR_DRBG on top of PSA. The condition as currently implemented is wrong when PSA is not using CTR_DRBG. It can use `mbedtls_psa_external_get_random` or HMAC_DRBG instead. There's no CI failure because we currently always use CTR_DRBG for testing except in dedicated components.

**So no, please do not try to calculate the expected hits due to the internal RNG. They really need to be excluded from the count.**

The only way I can think of is to somehow detect when the driver is being called to process the RNG.

One way to do that would be to add a test-only observable when `ctr_drbg.c` calls `psa_cipher_update`:
```
#if defined(MBEDTLS_TEST_HOOKS)
    ++mbedtls_ctr_drbg_calls_to_psa_cipher_update;
#endif
```
And then test code can subtract `mbedtls_ctr_drbg_calls_to_psa_cipher_update`. Though this would need to be made thread-safe.

A better way to do that would be to detect when the driver is being called on an internal key. So there should be an internal interface to determine which keys are internal?

> Random thinking: do you think it could make sense to give this AES key used from CTR_DRBG an ID which is lower than MBEDTLS_PSA_KEY_ID_BUILTIN_MIN?

That would require a lot of changes to the slot management code. I have a bad feeling about adding such complexity there.